pkg/cmd/roachtest/tests/mixed_version_cdc.go

```diff
@@ -244,6 +265,50 @@ func (cmvt *cdcMixedVersionTester) timestampResolved() {
 	}
 }
 
+// cdcDBConn is the wrapper passed to the FingerprintValidator. The
+// goal is to ensure that database checks by the validator do not
+// happen while we are running an upgrade. We used to retry database
+// calls in the validator, but that logic adds complexity and does not
+// help in testing the changefeed's correctness
+func (cmvt *cdcMixedVersionTester) cdcDBConn(
+	getConn func(int) *gosql.DB,
+) func(func(*gosql.DB) error) error {
+	return func(f func(*gosql.DB) error) error {
+		cmvt.crdbUpgrading.Lock()
+		defer cmvt.crdbUpgrading.Unlock()
+
+		node := cmvt.crdbNodes.RandNode()[0]
+		return f(getConn(node))
+	}
+}
+
+// crdbUpgradeStep is a wrapper to steps that upgrade the cockroach
+// binary running in the cluster. It makes sure we hold exclusive
+// access to the `crdbUpgrading` lock while the upgrade is in process
+func (cmvt *cdcMixedVersionTester) crdbUpgradeStep(step versionStep) versionStep {
```

Comment:
We should call `runCDCMixedVersion` somethin like `runCDCMixedVersionPaused` to make it clear that we're not testing the running case.  We'd want to be testing this paused case as well anyway. Could also be good to make this `crdbUpgradeStep` explicitly somethin like `crdbUpgradeStepPaused`

Reply:
I can do that in a separate PR, properly refactoring this test to take parameters. For this one I'd like to just address #87251.

>  We'd want to be testing this paused case as well anyway

Is this what we recommend customers to do? Or something that you know customers do anyway when they are upgrading? Curious about what's the more common approach to deal with changefeeds during upgrades.